rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,110 @@
+
+The appropriate on call responder can then click into the provided link and approve or deny the access request.
+
+For auto approval of certain access requests the access request will be auto approved if the requesting user is on-call in one of the services provided in request annotation. (Provided auto approval is enabled in the config)
+
+Once an access request has been approved or denied the plugin will add a note to the alert and close the relevant alert tied to that access request.
+
+## Implementation details
+In this section we will take a look at how the plugin will interact with the Opsgenie API.
+
+### Authorization
+
+The plugin will use the API key provided in the Opsgenie config file when interacting with the Opsgenie API. This API key will be included in the headers of the requests made.
+
+```
+Header Key: Authorization
+Header Value: genieKey $apiKey
+```
+
+### Creating alerts
+When the Opsgenie plugin creates alerts for incoming access requests the Create alert request will be of the form
+
+```
+{
+	"message": "Access request from <someuser>",
+	"description":"<someuser> requested permissions for roles <someroles> on Teleport at <someTime>.
+ 	Reason: <someReason>
+ 	To approve or deny the request, proceed to <link to the access request>",
+	"responders":[
+    	....
+	],
+	"tags": ["TeleportAccessRequest"],
+	"priority":"<somePriority>"
+}
+```
+
+On every review of the access request the alert created in Opsgenie will have a note added to it using the ‘Add note to alert’ endpoint. Then the alert will be closed using the ‘Close alert’ endpoint once the alert has either been approved or denied.
+
+```
+<Reviewer> reviewed the request at <someTime>.
+Resolution: <StateEmojiAsUsedByExistingPLugins> <State>.
+Reason: <Reason>
+```
+
+### Auto approval
+
+To check if the requesting user of a request is currently on-call the ‘Who is on call API’s ‘Get on calls’ endpoint will be used. 'https://<configured-opsgenie-address>/v2/schedules/<SheduleName>/on-calls?scheduleIdentifierType=name'
+
+Similar to the existing Pagerduty plugin for auto-approval to work, the user creating an Access Request must have a Teleport username that is also the email address associated with an Opsgenie account.
+
+Access requests will be mapped to Opsgenie alerts by including the Access request ID in the tags field of the note. 
+
+Shared code between the teleport-plugins found in lib is not too extensive and the simplest method to handle this when adding the Opsgenie plugin would be to simply duplicate what is needed for now.
+
+### Note
+
+Opsgenie will only ever read Teleport state and never write it.
+
+## Security considerations
+
+Potential for users to get access requests auto approved if they can get themselves onto the current on call rotation.
+Since Teleport usernames are assumed to match the Opsgenie email address when checking on call there is potential for access requests to be auto approved unintentionally.
```

Comment:
But only for the roles they are allowed to request access to, right?
As an example
https://goteleport.com/docs/access-controls/access-requests/role-requests/#role-access-requests-setup
In this case, the user is only allowed to ask for the `dba` role

Reply:
Yes, this would only be a concern for roles they are able to create access requests for.